test/unit/metrics/test_extra_metrics.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import fairlearn.metrics as metrics
+
+import pytest
+
+# ==============================================
+# Specificity (aka True Negative Rate)
+
+
+def test_specificity_all_correct():
+    y_true = [0, 0, 0, 0, 1]
+    y_pred = [0, 0, 0, 0, 1]
+
+    result = metrics.specificity_score(y_true, y_pred)
+    assert result == 1
+
+
+def test_specificity_none_correct():
+    y_true = [0, 0, 0, 0, 1]
+    y_pred = [1, 1, 1, 1, 0]
+
+    result = metrics.specificity_score(y_true, y_pred)
+    assert result == 0
+
+
+def test_specificity_some_correct():
+    y_true = [0, 0, 0, 0, 1]
+    y_pred = [0, 0, 0, 1, 1]
+
+    result = metrics.specificity_score(y_true, y_pred)
+    assert result == 0.75
+
+
+def test_specificity_some_correct_with_false_negative():
+    y_true = [0, 0, 0, 0, 1]
+    y_pred = [0, 0, 1, 0, 0]
+
+    result = metrics.specificity_score(y_true, y_pred)
+    assert result == 0.75
+
+
+# ========================================
```

Comment:
if there should be a logical separation, can we just put it in a different file?

Reply:
If the files grow much bigger, certainly. I'm trying to balance keeping things separate with having lots of tiny files scattered all over the place.

Reply:
I've pulled `selection_rate` into its own file, since that is a bit more sophisticated.